frontend/src/__tests__/cypress/cypress/tests/mocked/hardwareProfiles/manageHardwareProfiles.cy.ts

```diff
@@ -0,0 +1,446 @@
+import { TolerationEffect, TolerationOperator } from '~/types';
+import { mockK8sResourceList } from '~/__mocks__/mockK8sResourceList';
+import { HardwareProfileModel } from '~/__tests__/cypress/cypress/utils/models';
+import { asProductAdminUser } from '~/__tests__/cypress/cypress/utils/mockUsers';
+import { mockHardwareProfile } from '~/__mocks__/mockHardwareProfile';
+import {
+  createHardwareProfile,
+  createNodeResourceModal,
+  createNodeSelectorModal,
+  createTolerationModal,
+  duplicateHardwareProfile,
+  editHardwareProfile,
+  editNodeResourceModal,
+  editNodeSelectorModal,
+  editTolerationModal,
+} from '~/__tests__/cypress/cypress/pages/hardwareProfile';
+
+type HandlersProps = {
+  isPresent?: boolean;
+};
+
+const initIntercepts = ({ isPresent = true }: HandlersProps) => {
+  cy.interceptK8s(
+    { model: HardwareProfileModel, ns: 'opendatahub', name: 'test-hardware-profile' },
+    isPresent
+      ? mockHardwareProfile({
+          namespace: 'opendatahub',
+          name: 'test-hardware-profile',
+          displayName: 'Test Hardware Profile',
+          description: 'Test description',
+          identifiers: [
+            {
+              displayName: 'RAM',
+              identifier: 'memory',
+              minCount: '2Gi',
+              maxCount: '5Gi',
+              defaultCount: '2Gi',
+            },
+            {
+              displayName: 'CPU',
+              identifier: 'cpu',
+              minCount: '1',
+              maxCount: '2',
+              defaultCount: '1',
+            },
+            {
+              identifier: 'nvidia.com/gpu',
+              displayName: 'GPU',
+              maxCount: 2,
+              minCount: 1,
+              defaultCount: 1,
+            },
+          ],
+          tolerations: [
+            {
+              key: 'nvidia.com/gpu',
+              operator: TolerationOperator.EXISTS,
+              effect: TolerationEffect.NO_SCHEDULE,
+            },
+          ],
+          nodeSelectors: [{ key: 'test-key', value: 'test-value' }],
+        })
+      : {
+          statusCode: 404,
+        },
+  );
+};
+
+describe('Manage Hardware Profile', () => {
+  beforeEach(() => {
+    asProductAdminUser();
+  });
+
+  it('create hardware profile', () => {
```

Comment:
Can you add test case for checking error when min is greater than max, as well as default? This test case has became too long, maybe you can add that in new case, upto you.